### DISCUSSION TITLE
Can't connect (self server connected) Rustdesk Client to a public Rustdesk client

### DISCUSSION THREAD

**Discussion by arladmin** [rustdesk/rustdesk#2559] [Q&A] [answered]
I am hosting Rustdesk on my server and have correspondingly, linked the client on my system to my server.

But, most of the times, i am in a scenario where i ask the other person to download Rustdesk client from the official website but then i'm unable to connect to them. 
Error: "ID not found"

What's the issue here and how to resolve it?

**Comment by jabo59** [comment]
Have you configured the "ID Server" and "Key" of your server in every client?

**Reply by arladmin** [reply]
No.
Why would i share my server's ID and Key with public/external users?

Also, if not for the security aspect, it would make a hugely negative user experience for end-users, who just need to receive a quick and temporary session, to have to enter the ID and Key. Most of these users are used to the seameless experiences provided by other similar tools, like Anydesk and Teamviewer.

**Reply by jabo59** [reply]
Then why did you set up an own rustdesk server? In your scenario it seems to me that you can go with a public server.

**Reply by arladmin** [reply]
1. To connect seamlessly with internal users
2. Based on Rustdesk's recommendation to have a faster connection performance

Didn't expect that to cause access issues. Also, didn't realise that it would be a one-or-the-other situation.

**Reply by jabo59** [reply]
Hm. so you have 2 use cases:
1) internal users
2) public users

If the internal users are in the same network as you, then a direct connection is feasible. In this case you don't need the rustdesk server. I suggest to remove your server and check if a direct connection can be established. Hint: Direct connections are not encrypted.

If you have configured your id server in your client, I believe you can only establish connection through this server or directly (if the configuration both client and server allows this).

**Reply by arladmin** [reply]
That's correct. 

But,
1. The connection with the internal users is over LAN as well as (non-VPN) internet
2. From my experience so far, it becomes an issue for every single user that i ask to install Rustdesk. These (internal/public) users might enter the Server info, to let me connect with them, but then also need to connect with other internal/public users, which they then can't.

So, in effect, the self-hosted ecosystem losses its charm as it's unusable in half of the use-cases. 
On the other hand, the public ecosystem has its own set of Cons (performance over time, security).

**Reply by jabo59** [reply]
I agree with your conclusion.

**Reply by arladmin** [reply]
I see 2 solutions, with my limited knowledge:

1. Have separate clients for the 2 ecosystems, and/or
2. Self-hosted server should be able to connect with external users

I would prefer the 2nd and i would assume, most people would prefer that too.

**Comment by pepe12311** [comment]
Regarding this topic, which I see is old. Has any solution been proposed?

Well, the direct connection is a bit faster, but there is the problem that if I need to connect to any other computer, I can't, because the client needs to be configured.

The only way I'm seeing is to have another Rustdesk on a virtual machine to use one or the other, which is a problem. It would be interesting if you could define each connection separately or even have it do it automatically.

**Reply by Satalink** [reply]
Try "NoMachine" .   I gave up on RustDesk

**Comment by rustdesk** [comment, marked as answer]
https://twitter.com/rustdesk/status/1721799106698396029

**Reply by QuimaxW** [reply]
The link is dead...

The full answer to this question is to put `<remoteid>@public` in the connect to box. I wasn't able to find documentation for this, but it works. I suspect it could also be used to connect via a different rustdesk server as well by changing the @public to @differentserverip, but I don't know this for sure.
